src/dbnode/persist/fs/files.go

```diff
@@ -1378,6 +1411,28 @@ func readAndValidate(
 	return buf[:n], nil
 }
 
+func read(filePath string) ([]byte, error) {
+	fd, err := os.Open(filePath) //nolint:gosec
+	if err != nil {
+		return nil, err
+	}
+	defer fd.Close() //nolint:errcheck,gosec
+
+	stat, err := os.Stat(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	size := int(stat.Size())
+	buf := make([]byte, size)
```

Comment:
Nit: this part of code is exactly the same as in `readAndValidate`. Might be worth extracting something like `func bufferForEntireFile(filePath string) ([]byte, error)`